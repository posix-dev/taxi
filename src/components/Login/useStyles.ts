import {createStyles} from "@material-ui/core";
import background from "../../background.png";

export const useStyles = createStyles({
    loginFormWrapper: {
        backgroundImage: `url(${background})`,
        height: 790
    },
    loginForm: {
        paddingLeft: 24,
        paddingRight: 24,
        paddingTop: 48
    },
    loginRegistrationLink: {
        fontSize: 14,
        textDecoration: "underline",
        color: "blue",
        cursor: "pointer"
    },
    loginSubmitBtn: {
        marginTop: 36,
        marginBottom: 48,
        minWidth: 160,
        minHeight: 46,
        fontSize: 18,
        fontWeight: 500,
        textTransform: "none"
    }
});