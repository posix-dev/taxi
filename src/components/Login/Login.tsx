import React, {SyntheticEvent} from 'react';

import logo from '../../logo-taxi-light.png';
import {Box, Button, Card, Grid, TextField, withStyles} from "@material-ui/core";
import CardContent from "@material-ui/core/CardContent";
import Typography from "@material-ui/core/Typography";
import {useStyles} from "./useStyles"
import AuthContext, {AuthConsumer} from "../App/AuthProvider";
import {Link} from "react-router-dom";

interface LoginProps {
    classes: any
}

interface LoginState {
    name: String,
    password: String
}

class Login extends React.Component<LoginProps, LoginState> {
    static contextType = AuthContext;

    constructor(props: LoginProps) {
        super(props);
        this.state = {name: "", password: ""};

        this.handleChangeName = this.handleChangeName.bind(this);
        this.handleChangePassword = this.handleChangePassword.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChangeName(e: React.ChangeEvent<HTMLInputElement>) {
        this.setState({name: e.target.value});
    }

    handleChangePassword(e: React.ChangeEvent<HTMLInputElement>) {
        this.setState({password: e.target.value});
    }

    handleSubmit(e: SyntheticEvent) {
        e.preventDefault();
        this.context.login(this.state.name, this.state.password);
    }

    render(): React.ReactElement {
        const {classes} = this.props;

        return (
            <AuthConsumer>
                {auth => (
                    <Grid className={classes.loginFormWrapper} container>
                        <Grid xs={7} alignItems={"center"} container item justify={"center"}>
                            <img src={logo} alt="Логотип компании"/>
                        </Grid>
                        <Grid xs={4} justify={"flex-end"} alignItems={"center"} container item>
                            <Card className={classes.loginForm}>
                                <CardContent>
                                    <Typography
                                        align="left"
                                        gutterBottom={true}
                                        variant="h3"
                                        component="h1">
                                        <Box fontWeight="fontWeightBold">Войти</Box>
                                    </Typography>
                                    <Typography
                                        align="left"
                                        gutterBottom={true}
                                        variant="h5"
                                        component="h2">
                                        <Box fontWeight="fontWeightLight">
                                            Новый пользователь?&nbsp;
                                            <Link
                                                className={classes.loginRegistrationLink}
                                                to="/registration">Зарегистрируйтесь</Link>
                                        </Box>
                                    </Typography>
                                    <form autoComplete="off">
                                        <TextField
                                            inputProps={{style: {fontSize: 18}, "data-testid": "login"}}
                                            InputLabelProps={{style: {fontSize: 18}}}
                                            fullWidth
                                            id="login"
                                            // data-testid="login"
                                            color="secondary"
                                            label="Имя пользователя"
                                            required
                                            onChange={this.handleChangeName}
                                            value={this.state.name}
                                            name={"login"}/>
                                        <br/>
                                        <div style={{marginTop: 28}}/>
                                        <TextField
                                            inputProps={{style: {fontSize: 18, minWidth: 370}}}
                                            InputLabelProps={{style: {fontSize: 18}}}
                                            id="password"
                                            data-testid="myPassword"
                                            label="Пароль"
                                            color="secondary"
                                            onChange={this.handleChangePassword}
                                            required type={"password"}
                                            name={"password"}/>
                                        <br/>
                                        <Grid container item justify={"flex-end"}>
                                            <Button
                                                className={classes.loginSubmitBtn}
                                                type="submit"
                                                data-testid="loginSubmit"
                                                variant="contained"
                                                color="primary"
                                                onClick={e => this.handleSubmit(e)}
                                            >Войти</Button>
                                        </Grid>
                                    </form>
                                </CardContent>
                            </Card>
                        </Grid>
                    </Grid>
                )}
            </AuthConsumer>
        )
    }
}

export default withStyles(useStyles, {withTheme: true})(Login);