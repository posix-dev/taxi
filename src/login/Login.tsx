import React, {SyntheticEvent} from 'react';

import logo from '../logo-taxi-light.png';
import {Box, Button, Card, Grid, Link, TextField, withStyles} from "@material-ui/core";
import CardContent from "@material-ui/core/CardContent";
import Typography from "@material-ui/core/Typography";
import {NavigateConsumer} from '../app/NavigateContext'
import {useStyles} from "./useStyles"
import AuthContext, {AuthConsumer} from "../app/AuthProvider";

interface LoginProps {
    classes: any
}

interface LoginState {
    name: String,
    password: String
}

class Login extends React.Component<LoginProps, LoginState> {
    static contextType = AuthContext;

    constructor(props: LoginProps) {
        super(props);
        this.state = {name: "", password: ""};

        this.handleChangeName = this.handleChangeName.bind(this);
        this.handleChangePassword = this.handleChangePassword.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChangeName(e: React.ChangeEvent<HTMLInputElement>) {
        this.setState({name: e.target.value});
    }

    handleChangePassword(e: React.ChangeEvent<HTMLInputElement>) {
        this.setState({password: e.target.value});
    }

    handleSubmit(e: SyntheticEvent) {
        e.preventDefault();
        this.context.login(this.state.name, this.state.password);
    }

    render(): React.ReactElement {
        const {classes} = this.props;

        return (
            <NavigateConsumer>
                {nav => (
                    <AuthConsumer>
                        {auth => (
                            <Grid className={classes.loginFormWrapper} container>
                                <Grid xs={7} alignItems={"center"} container item justify={"center"}>
                                    <img src={logo} alt="Логотип компании"/>
                                </Grid>
                                <Grid xs={4} justify={"flex-end"} alignItems={"center"} container item>
                                    <Card className={classes.loginForm}>
                                        <CardContent>
                                            <Typography
                                                align="left"
                                                gutterBottom={true}
                                                variant="h3"
                                                component="h1">
                                                <Box fontWeight="fontWeightBold">Войти</Box>
                                            </Typography>
                                            <Typography
                                                align="left"
                                                gutterBottom={true}
                                                variant="h5"
                                                component="h2">
                                                <Box fontWeight="fontWeightLight">
                                                    Новый пользователь?&nbsp;
                                                    <Link
                                                        className={classes.loginRegistrationLink}
                                                        onClick={(e: SyntheticEvent) => {
                                                            e.preventDefault();
                                                            nav.navigate("registration");
                                                        }}
                                                        href="#">Зарегистрируйтесь</Link>
                                                </Box>
                                            </Typography>
                                            <form autoComplete="off">
                                                <TextField
                                                    inputProps={{style: {fontSize: 18}}}
                                                    InputLabelProps={{style: {fontSize: 18}}}
                                                    fullWidth
                                                    id="login"
                                                    data-testid="login"
                                                    color="secondary"
                                                    label="Имя пользователя"
                                                    required
                                                    onChange={this.handleChangeName}
                                                    name={"login"}/>
                                                <br/>
                                                <div style={{marginTop: 28}}/>
                                                <TextField
                                                    inputProps={{style: {fontSize: 18, minWidth: 370}}}
                                                    InputLabelProps={{style: {fontSize: 18}}}
                                                    id="password"
                                                    data-testid="myPassword"
                                                    label="Пароль"
                                                    color="secondary"
                                                    onChange={this.handleChangePassword}
                                                    required type={"password"}
                                                    name={"password"}/>
                                                <br/>
                                                <Grid container item justify={"flex-end"}>
                                                    <Button
                                                        className={classes.loginSubmitBtn}
                                                        type="submit"
                                                        data-testid="loginSubmit"
                                                        variant="contained"
                                                        color="primary"
                                                        onClick={e => this.handleSubmit(e)}
                                                    >Войти</Button>
                                                </Grid>
                                            </form>
                                        </CardContent>
                                    </Card>
                                </Grid>
                            </Grid>
                        )}
                    </AuthConsumer>
                )}
            </NavigateConsumer>
        )
    }
}

export default withStyles(useStyles, {withTheme: true})(Login);