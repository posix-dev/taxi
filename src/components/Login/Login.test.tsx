import React from 'react';
import renderer, {act} from 'react-test-renderer';
import {render, cleanup, fireEvent} from '@testing-library/react';
import Login from './Login';

describe('Login', () => {
    const {getByTestId} = render(<Login/>);
    describe('init renders correctly', () => {
        expect(getByTestId('login')).toHaveTextContent("Имя пользователя");
        expect(getByTestId('myPassword')).toHaveTextContent("Пароль");
        expect(getByTestId('loginSubmit')).toBeTruthy();
        expect(getByTestId('loginSubmit')).toHaveAttribute("data-testid");

        it('snapshot test passed', () => {
            const tree = renderer.create(<Login/>).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });

    describe('check init state', () => {
        it('expect correct change state when we fill email', () => {
            const mockedFunction = jest.fn();
            const input = getByTestId('login');
            input.onchange = mockedFunction;
            act(() => {
                fireEvent.change(input, {target: {value: 'My new value'}});
            });
            expect(mockedFunction).toHaveBeenCalled();
        });
    });
});

afterEach(cleanup);
