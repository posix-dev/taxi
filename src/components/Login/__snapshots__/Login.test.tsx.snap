// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login init renders correctly snapshot test passed 1`] = `
<div
  className="MuiGrid-root Login-loginFormWrapper-1 MuiGrid-container"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-center MuiGrid-grid-xs-7"
  >
    <img
      alt="Логотип компании"
      src="logo-taxi-light.png"
    />
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-flex-end MuiGrid-grid-xs-4"
  >
    <div
      className="MuiPaper-root MuiCard-root Login-loginForm-2 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiCardContent-root"
      >
        <h1
          className="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom MuiTypography-alignLeft"
        >
          <div
            className="MuiBox-root MuiBox-root-7"
          >
            Войти
          </div>
        </h1>
        <h2
          className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom MuiTypography-alignLeft"
        >
          <div
            className="MuiBox-root MuiBox-root-8"
          >
            Новый пользователь? 
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover Login-loginRegistrationLink-3 MuiTypography-colorPrimary"
              href="#"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
            >
              Зарегистрируйтесь
            </a>
          </div>
        </h2>
        <form
          autoComplete="off"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            data-testid="login"
          >
            <label
              className="MuiFormLabel-root MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
              data-shrink={false}
              htmlFor="login"
              id="login-label"
              style={
                Object {
                  "fontSize": 18,
                }
              }
            >
              Имя пользователя
              <span
                aria-hidden={true}
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorSecondary MuiInput-colorSecondary MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                id="login"
                name="login"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                style={
                  Object {
                    "fontSize": 18,
                  }
                }
                type="text"
                value=""
              />
            </div>
          </div>
          <br />
          <div
            style={
              Object {
                "marginTop": 28,
              }
            }
          />
          <div
            className="MuiFormControl-root MuiTextField-root"
            data-testid="myPassword"
          >
            <label
              className="MuiFormLabel-root MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
              data-shrink={false}
              htmlFor="password"
              id="password-label"
              style={
                Object {
                  "fontSize": 18,
                }
              }
            >
              Пароль
              <span
                aria-hidden={true}
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorSecondary MuiInput-colorSecondary MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                id="password"
                name="password"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                style={
                  Object {
                    "fontSize": 18,
                    "minWidth": 370,
                  }
                }
                type="password"
              />
            </div>
          </div>
          <br />
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-flex-end"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained Login-loginSubmitBtn-4 MuiButton-containedPrimary"
              data-testid="loginSubmit"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="submit"
            >
              <span
                className="MuiButton-label"
              >
                Войти
              </span>
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
